### PR TITLE
[chore] arm64 build

### PR DESCRIPTION
**Pull request by atoulme**
This PR adds a linux/arm64 build to the build of the collector, so it may support the goal of #9731

### CODE REVIEW

File: processor/memorylimiterprocessor/internal/mock_exporter.go [hunk 1]
```diff
@@ -24,44 +25,48 @@ func (e *MockExporter) Capabilities() consumer.Capabilities {
...
 }
 
 func NewMockExporter() *MockExporter {
-	return &MockExporter{}
+	return &MockExporter{
+		destAvailable:     atomic.Bool{},
+		acceptedLogCount:  atomic.Int64{},
+		deliveredLogCount: atomic.Int64{},
+		Logs:              consumertest.LogsSink{},
+	}
```
Comment (dmitryax) [inline processor/memorylimiterprocessor/internal/mock_exporter.go:71]: Are these changes required for the actuated? Can you move them to a separate PR?

Comment (atoulme) [reply processor/memorylimiterprocessor/internal/mock_exporter.go:71]: they are as the build on arm was failing, and it made sense to have them to make sure actuated passed. I can split the PR in 2, sure.

Comment (atoulme) [reply processor/memorylimiterprocessor/internal/mock_exporter.go:71]: https://github.com/open-telemetry/opentelemetry-collector/pull/9733 is open to split.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -137,10 +137,18 @@ jobs:
...
     strategy:
       matrix:
+        runner: [ubuntu-latest, actuated-arm64-4cpu-4gb]
         go-version: ["~1.22", "~1.21.8"] # 1.20 needs quotes otherwise it's interpreted as 1.2
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.runner }}
     needs: [setup-environment]
     steps:
+      - name: Set up arkade
+        uses: alexellis/setup-arkade@master
```
Comment (mx-psi) [inline .github/workflows/build-and-test.yml:146]: We should pin this to a tag or, in the absence of tags, to a specific commit

**Review by mx-psi — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (atoulme) [reply .github/workflows/build-and-test.yml:146]: That's fair ; updated.

**Comment by atoulme** [comment]
@open-telemetry/collector-approvers anyone interested in reviewing this PR?

**Review by dmitryax — APPROVED** [review]
